llvm/labs/lab2/vyunov_danila/magic_inline.cpp

```diff
@@ -0,0 +1,121 @@
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/DerivedTypes.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/Instruction.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/Value.h"
+#include "llvm/Passes/PassBuilder.h"
+#include "llvm/Passes/PassPlugin.h"
+#include "llvm/Support/Casting.h"
+#include "llvm/Transforms/Utils/BasicBlockUtils.h"
+#include "llvm/Transforms/Utils/Cloning.h"
+#include "llvm/Transforms/Utils/ValueMapper.h"
+#include <map>
+#include <vector>
+
+namespace {
+llvm::BasicBlock *splitBlockBefore(llvm::BasicBlock *Old,
+                                   llvm::Instruction *SplitPt,
+                                   llvm::DomTreeUpdater *DTU = nullptr,
+                                   llvm::LoopInfo *LI = nullptr,
+                                   llvm::MemorySSAUpdater *MSSAU = nullptr,
+                                   const llvm::Twine &BBName = "",
+                                   bool Before = true) {
+  return llvm::SplitBlock(Old, SplitPt, DTU, LI, MSSAU, BBName, Before);
```

Comment:
May we remove this function and just use `SplitBlock` call?

Reply:
We can delete it if there are no special requirements that our function fulfills